libvirt/resource_libvirt_network.go

```diff
@@ -320,6 +342,12 @@ func resourceLibvirtNetworkUpdate(d *schema.ResourceData, meta interface{}) erro
 		return fmt.Errorf("Error updating DNS hosts for network %s: %s", networkName, err)
 	}
 
+	// FIXME assuming only one dhcp address range can be active
+	err = updateDHCPHosts(d, 0, network)
```

Comment:
Hi. Thanks for the PR.

The implementation looks good. I am still unsure where we should merge this, and this line highlights why.

The original design of _terraform-provider-libvirt_ was to act as a simple DigitalOcean-like Cloud with Libvirt. You start a VM easily, and therefore the defaults where a single interface, NAT and DHCP.

It wasn't until people used this for testing, especially Kubernetes, when complex network setups started to arise.

At that time, networking was extended so that `addresses` is a plain array of addresses, therefore `dhcp`continues being a global block instead of one associated by address.

This design is what also has me unsure of merging https://github.com/dmacvicar/terraform-provider-libvirt/pull/725, in the past NAT was default, and creating a proper `forward` block means NAT is not longer the default.

I am not sure what the is the right thing to do. A new network schema that specifies `ip` blocks and their own `forward`and `dhcp`blocks would prevent this kind of hacks. At the same time, it would make a default schema more complex than it is today. The libvirt schema for network is deeply nested.

Given that we are seeing increasingly complex network configurations, I am inclined for a full network resource re-write, following the libvirt-schema almost 1:1, and add this kind of features there. After all, both network resources could coexist for a while (eg `libvirt_network_v2`).